src/comps/Header.tsx

```diff
@@ -2,52 +2,176 @@ import Image from "next/image";
+          userData.profile.walletProvider === WALLET_PROVIDER_NAMES.xverse
+        ) {
+          // TODO: support xverse wallet
+          _events.push({
+            id: _events.length + 1 + "",
+            type: NotificationStatusType.ERROR,
+            title: `Xverse is not supported yet`,
+          });
+        } else {
+          // wallet not supported
+          _events.push({
+            id: _events.length + 1 + "",
+            type: NotificationStatusType.ERROR,
+            title: `Wallet not supported`,
+          });
+        }
+      } else if (stacksNetwork.chainId === STACKS_MAINNET.chainId) {
+        // main net wallet address
+        if (
+          userData.profile.walletProvider === WALLET_PROVIDER_NAMES.leather ||
+          userData.profile.walletProvider === WALLET_PROVIDER_NAMES.leather
+        ) {
+          stxWalletAddress = userData.profile.stxAddress.mainnet;
+        } else if (
+          userData.profile.walletProvider === WALLET_PROVIDER_NAMES.xverse
+        ) {
+          // TODO: support xverse wallet
+          _events.push({
+            id: _events.length + 1 + "",
+            type: NotificationStatusType.ERROR,
+            title: `Xverse is not supported yet`,
+          });
+        } else {
+          // wallet not supported
+          _events.push({
+            id: _events.length + 1 + "",
+            type: NotificationStatusType.ERROR,
+            title: `Wallet not supported`,
+          });
+        }
+      } else {
+        // stacks env not supported
+        _events.push({
+          id: _events.length + 1 + "",
+          type: NotificationStatusType.ERROR,
+          title: `Stacks env not supported`,
+        });
+      }
+
+      if (stxWalletAddress) {
+        setWalletAddress(stxWalletAddress);
+      } else {
+        // could not get wallet address
+        _events.push({
+          id: _events.length + 1 + "",
+          type: NotificationStatusType.ERROR,
+          title: `Could not get wallet address`,
+        });
+      }
+
       setEvents(_events);
     }
   }, []);
 
+  useEffect(() => {
+    if (isConnected && walletAddress) {
+      fetchSignerAggregateAddress();
+    }
+  }, [isConnected, walletAddress]);
+
+  const fetchSignerAggregateAddress = async () => {
+    try {
+      const network =
+        stacksNetwork.chainId === STACKS_TESTNET.chainId
+          ? STACKS_TESTNET
+          : STACKS_MAINNET;
+      const readOnlyHelperRes = await readOnlyHelper({
+        stacksNetwork: network,
+        walletAddress: walletAddress || "",
+        functionName: "get-current-aggregate-pubkey",
+      });
+
+      const uint8FormatRes = hexToUint8Array(readOnlyHelperRes);
+
+      // remove the first byte of test
+      const xOnlyuInt = uint8FormatRes.slice(1);
+
+      // convert uint8array to hex
+      const xOnlyHex = uint8ArrayToHexString(xOnlyuInt);
+
+      setSignerPubkey(xOnlyHex);
+    } catch (err) {
+      console.error(err);
+    }
+  };
```

Comment:
Maybe this would be a good place to integrate a retry logic or use react-query and instead of console.err you could use toast here

Reply:
Here would be a good place to use a server action